test/adapters/optimisticAuctionRebalanceExtension.spec.ts

```diff
@@ -0,0 +1,757 @@
+  let hexStr = "";
+
+  for (let i = 0; i < buffer.length; i++) {
+    const hex = (buffer[i] & 0xff).toString(16);
+    hexStr += hex.length === 1 ? "0" + hex : hex;
+  }
+
+  return hexStr;
+}
+
+// Base58 decoding function (make sure you have a proper Base58 decoding function)
+function base58ToHexString(base58String: string) {
+  const bytes = base58.decode(base58String); // Decode base58 to a buffer
+  const hexString = bufferToHex(bytes.slice(2)); // Convert buffer to hex, excluding the first 2 bytes
+  return "0x" + hexString;
+}
+
+describe("OptimisticAuctionRebalanceExtension", () => {
+  let owner: Account;
+  let methodologist: Account;
+  let operator: Account;
+
+  let setV2Setup: SetFixture;
+
+  let deployer: DeployHelper;
+  let setToken: SetToken;
+  let baseManager: BaseManager;
+
+  let auctionRebalanceExtension: OptimisticAuctionRebalanceExtension;
+
+  let priceAdapter: ConstantPriceAdapter;
+
+  let optimisticOracleV3Mock: OptimisticOracleV3Mock;
+
+  let optimisticOracleV3MockUpgraded: OptimisticOracleV3Mock;
+
+  let collateralAsset: StandardTokenMock;
+
+  let useAssetAllowlist: boolean;
+  let allowedAssets: Address[];
+
+  before(async () => {
+    [owner, methodologist, operator] = await getAccounts();
+
+    deployer = new DeployHelper(owner.wallet);
+
+    setV2Setup = getSetFixture(owner.address);
+    await setV2Setup.initialize();
+
+    priceAdapter = await deployer.setV2.deployConstantPriceAdapter();
+    optimisticOracleV3Mock = await deployer.mocks.deployOptimisticOracleV3Mock();
+    optimisticOracleV3MockUpgraded = await deployer.mocks.deployOptimisticOracleV3Mock();
+    collateralAsset = await deployer.mocks.deployStandardTokenMock(owner.address, 18);
+
+    await setV2Setup.integrationRegistry.addIntegration(
+      setV2Setup.auctionModule.address,
+      "ConstantPriceAdapter",
+      priceAdapter.address,
+    );
+
+    useAssetAllowlist = false;
+    allowedAssets = [];
+
+    setToken = await setV2Setup.createSetToken(
+      [setV2Setup.dai.address, setV2Setup.wbtc.address, setV2Setup.weth.address],
+      [ether(100), bitcoin(0.01), ether(0.1)],
+      [setV2Setup.auctionModule.address, setV2Setup.issuanceModule.address],
+    );
+
+    await setV2Setup.issuanceModule.initialize(setToken.address, ADDRESS_ZERO);
+
+    baseManager = await deployer.manager.deployBaseManager(
+      setToken.address,
+      operator.address,
+      methodologist.address,
+    );
+    baseManager = baseManager.connect(operator.wallet);
+    await setToken.setManager(baseManager.address);
+
+    auctionRebalanceExtension = await deployer.extensions.deployOptimisticAuctionRebalanceExtension(
+      baseManager.address,
+      setV2Setup.auctionModule.address,
+      useAssetAllowlist,
+      allowedAssets,
+    );
+    auctionRebalanceExtension = auctionRebalanceExtension.connect(operator.wallet);
+  });
+
+  addSnapshotBeforeRestoreAfterEach();
+
+  describe("#constructor", () => {
```

Comment:
let's simplify the constructor tests, see https://github.com/IndexCoop/index-coop-smart-contracts/blob/4aa9028018a2ac533b60b55bf01c03185285616b/test/adapters/auctionRebalanceExtension.spec.ts#L83